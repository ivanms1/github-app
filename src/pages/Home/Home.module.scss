.Home {
  align-items: center;
  display: flex;
  flex-direction: column;
  gap: 2rem 0;
  height: 100%;
  margin-top: 5rem;

  & .Title {
    font-size: 3rem;
    font-weight: bold;
    text-align: center;
  }

  & .WelcomeSVG {
    max-height: 400px;
  }
}
