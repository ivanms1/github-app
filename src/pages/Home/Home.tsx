import React from "react";
import { signIn } from "next-auth/react";
import { Button, Text } from "@mantine/core";
import Head from "next/head";

import WelcomeIllustration from "@/assets/icons/welcome.svg";

import styles from "./Home.module.scss";

function Home() {
  return (
    <div className={styles.Home}>
      <Head>
        <title>Github App</title>
      </Head>
      <Text className={styles.Title}>Welcome to my Github App</Text>
      <WelcomeIllustration className={styles.WelcomeSVG} />
      <Button size="xl" onClick={() => signIn("github")}>
        Sign in
      </Button>
    </div>
  );
}

export default Home;
