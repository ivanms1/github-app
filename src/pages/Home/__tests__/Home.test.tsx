import { render } from "@testing-library/react";

import Home from "../Home";

test("it should render the home page", () => {
  const { container } = render(<Home />);

  expect(container).toMatchSnapshot();
});
