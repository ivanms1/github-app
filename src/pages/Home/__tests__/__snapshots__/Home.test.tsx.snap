// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render the home page 1`] = `
<div>
  <div
    class="Home"
  >
    <div
      class="mantine-Text-root Title mantine-qls5vf"
    >
      Welcome to my Github App
    </div>
    <svg
      class="WelcomeSVG"
      data-file-name="SvgWelcome"
    />
    <button
      class="mantine-Button-filled mantine-Button-root mantine-1kkl9mh"
      type="button"
    >
      <div
        class="mantine-3xbgk5 mantine-Button-inner"
      >
        <span
          class="mantine-qo1k2 mantine-Button-label"
        >
          Sign in
        </span>
      </div>
    </button>
  </div>
</div>
`;
